relay-dynamic-config/src/global.rs

```diff
@@ -70,6 +71,25 @@ impl GlobalConfig {
             ErrorBoundary::Ok(f) => Some(f),
         }
     }
+
+    /// Modifies the global config after deserialization.
+    ///
+    /// - Adds hard-coded groups to metrics extraction configs.
+    pub fn normalize(&mut self) {
+        if let ErrorBoundary::Ok(config) = &mut self.metric_extraction {
+            for (group_name, metrics) in defaults::hardcoded_span_metrics() {
+                // We only define these groups if they haven't been defined by the upstream yet.
+                // This ensures that the innermost Relay always defines the metrics.
+                if let Entry::Vacant(entry) = config.groups.entry(group_name) {
+                    entry.insert(MetricExtractionGroup {
+                        is_enabled: false, // must be enabled via project config
+                        metrics,
+                        tags: Default::default(),
+                    });
+                }
+            }
+        }
+    }
```

Comment:
Not a fan of these `normalize` etc methods which need to be called and modify the state of the object. Can we make this a separate type which just adds the defaults? Or just directly move everthing to Sentry.

It's just so easy to forget or miss.